recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,418 @@
+            self.requires("libcurl/7.74.0")
+
+    def package_id(self):
+        for hl_comp in self._internal_requirements.keys():
+            if getattr(self.options, hl_comp):
+                for internal_requirement in self._internal_requirements[hl_comp]:
+                    setattr(self.info.options, internal_requirement, True)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+        # Keeping these unused folders make the source copy fail on Windows
+        tools.rmdir(os.path.join(self._source_subfolder, "code-generation", "generator"))
+        tools.rmdir(os.path.join(self._source_subfolder, "aws-cpp-sdk-core-tests"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        build_only = ["core"]
+        for sdk in self._sdks:
+            if getattr(self.options, sdk):
+                build_only.append(sdk)
+        self._cmake.definitions["BUILD_ONLY"] = ";".join(build_only)
+
+        self._cmake.definitions["BUILD_DEPS"] = False
+        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
+        self._cmake.definitions["ENABLE_TESTING"] = False
+        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
+
+        self._cmake.definitions["MINIMIZE_SIZE"] = self.options.min_size
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in self.settings.compiler.runtime
+
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "AWSSDK"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "AWSSDK"
+        self.cpp_info.names["cmake_find_package"] = "AWS"
+        self.cpp_info.names["cmake_find_package_multi"] = "AWS"
+        self.cpp_info.components["core"].names["cmake_find_package"] = "aws-sdk-cpp-core"
+        self.cpp_info.components["core"].names["cmake_find_package_multi"] = "aws-sdk-cpp-core"
+        self.cpp_info.components["core"].names["pkg_config"] = "aws-sdk-cpp-core"
+        self.cpp_info.components["core"].libs = ["aws-cpp-sdk-core"]
+        self.cpp_info.components["core"].requires = ["aws-c-event-stream::aws-c-event-stream-lib"]
+
+        for sdk in self._sdks:
+            if getattr(self.options, sdk):
```

Comment:
and due to https://github.com/conan-io/conan-center-index/pull/4401#discussion_r600682618, you can't trust this condition.